### ISSUE TITLE
Clean way to `mv` a subdataset (not clone) to a different parent dataset/folder

### ISSUE THREAD

**Issue by dorianps** [datalad/datalad#6710]
### What is the problem?

Hope everyone is well.

I have a big dataset created by a colleague, and I need to move it to as a subdataset in a different folder. Cloning is out of the question because the content is too big. I have done this in the past by simply moving the entire folder under the new parent dataset. After that, I have tried editing `.gitmodules` and `.git/config` and commit, but the new folder seems still untracked. Currently the following steps work (based on force creating a subdataset in the parent), but this is clearly not a clean solution. Something happens during the subdataset creation and I need to save the entire tree (hint: there are many subdatasets nested in the dataset I am moving). If I don't save the entire tree, there is always something that shows as modified downstream.
```

1. Sudo copy the dataset in the new location:
	e.g. sudo mv /DATA/sdas/IMAGE-HD/IMAGE-HD/data/processed/freesurfer ./

2. Take ownership of folder with sudo:
	e.g. sudo chown -R dorian:dorian freesurfer

3. Change permissions to appropriate ones
	e.g. chmod -R -750 freesurfer
	(keep playing with commands if there are permission denied issues)

4. Force create a subdataset in datalad:
	e.g., datalad create -d . --force freesurfer

5. Save all changes, but be sure to use recursive (there is some changes inside the subdatasets caused by the force creation inside the parent dataset that will need to be saved)
	e.g., datalad save --recursive

6. Make sure there is nothing to save
	e.g., datalad status
```

This is what happens if I don't do a full recursive save:
```
[dorian@mri MasterParent]$ datalad status
  deleted: freesurfer (dataset)
 modified: freesurfer-aggregated (dataset)
[dorian@mri MasterParent]$ datalad save
delete(ok): freesurfer (file)
save(ok): . (dataset)
action summary:
  delete (ok: 1)
  save (ok: 1)
[dorian@mri MasterParent]$ datalad status
 modified: freesurfer-aggregated (dataset)
[dorian@mri MasterParent]$ git status
On branch master
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)
  (commit or discard the untracked or modified content in submodules)

        modified:   freesurfer-aggregated (modified content)

no changes added to commit (use "git add" and/or "git commit -a")
[dorian@mri MasterParent]$ datalad save freesurfer-aggregated
add(ok): .gitattributes (file)
add(ok): .gitmodules (file)
add(ok): CHANGELOG.md (file)
add(ok): README.md (file)
add(ok): code/.gitattributes (file)
add(ok): code/README.md (file)
add(ok): code/curated_LUT.tsv (file)
save(ok): freesurfer-aggregated (dataset)
action summary:
  add (ok: 7)
  save (ok: 1)
[dorian@mri MasterParent]$ datalad status
 modified: freesurfer-aggregated (dataset)
```

Can anyone provide a full solution how to do this cleanly? I had [the same question](https://github.com/datalad/datalad/issues/4051) 2 years ago, but so far it does not work just to edit `git/config`.

### What steps will reproduce the problem?

Move an existing datalad dataset from one location to another under a parent dataset, and try to save it (register it) in the parent dataset.

### Datalad information

datalad 0.15.4

### Additional context

Linux Centos 7.9

### Have you had any success using DataLad before?

Yup

**Comment by yarikoptic** [comment]
if it is already a dataset, AFAIK you don't need `create`, you just need  (dropping all permissions specifics)
```
mv /DATA/sdas/IMAGE-HD/IMAGE-HD/data/processed/freesurfer ./
datalad save -m "adding precious freesurfer subdataset from sdas" -d . ./freesurfer  # which would add it as a submodule
```
then, if you have it hosted already somewhere you might want to adjust `.gitmodules` to provide better `url`  for it.

**Comment by yarikoptic** [comment]
oh -- and you better never do `chmod -R -750 freesurfer` on a repo with git-annex -- you are removing protection git-annex does on those files under .git/annex/objects.  Changing owner should be sufficient (unless you are also to change "shared" setup of the git/git-annex repo).

*hint*:  there is +X attribute for `chmod` so that `chmod g+rX` would make file executable (the `1` in numeric way to specify) only if it was executable for owner IIRC. I.e. it would not make your non-executable files executable for group, but would make directories executable.

**Comment by dorianps** [comment]
Ok, thanks @yarikoptic . I thought I did what you suggest before, but I started seeing messages like `adding file X to dataset` or `adding folder Y to dataset`. After that, I ran `datalad subdatasets` and there was nothing showing. I thought the folder was being commited in the parent dataset instead of being considered a subdataset, and I rolled back the commit and followed another route. But I will test more your scenario later and see if that is working. Will re-open if needed.
